### ISSUE TITLE
Glossary

### ISSUE THREAD

**Issue by 3sanket3** [reactjs/gu.reactjs.org#6]
### The words that should be transliterated.
> Whose English pronunciation are generally familiar to Gujarati reader

| Words    | Transliteration |
| -------- | --------------- |
| Application | એપ્લિકેશન |
| method | મેથડ |
| implement | ઈમ્પ્લીમેન્ટ |
| interface | ઇન્ટરફેસ |
| tutorial | ટ્યૂટોરિઅલ |
| value | વેલ્યુ |
| painless | પેઇનલેસ |
| invoke | ઈન્વોક  |

### The words needs to be kept as it is:

> Programming Jargons

| Words    |
| -------- |
| Elements |
| Component(s) |
| server |
| technology stack |
| Node |
| render |
| Encapsulated |
| state |
| UI |
| logic |
| templates |
|JavaScript|
|libraries| 
|frameworks|
|Plugins|
|Stateful|
|User|

### The words needs to be translated:

| Words    | Transliteration |
| -------- | --------------- |
|  Docs|          માર્ગદર્શિકા         |

**Comment by 3sanket3** [comment]
From my perspective and initial thoughts. I would love to hear your suggestions to improve the same.

**Comment by 3sanket3** [comment]
Instead of "કેમ છો?"

>Hello world should be "નમસ્તે " 
by @Str1kemeister

I think "કેમ છો?" gives a more Gujarati touch. isn't નમસ્તે a kind of a Hindi word?

**Comment by Str1kemeister** [comment]
Namastey may sounds like a hindi word but I think most appropriate way to say hello.
Google translate is also aiming at namastay for hello in Gujarati

**Comment by ron4ex** [comment]
If we go for literal translation of Hello then you're right @Str1kemeister, નમસ્તે is better. However the iconic phrase of "Hello world" has been used as a form of _greeting_ since the beginning. So in terms of _greetings_, I don't think there's a more iconic phrase than કેમ છો in Gujarati. Perhaps we can take a bit of liberty in this case and not go for the literal translation. Should we add દુનિયા at the end?

**Comment by sonipreshita** [comment]
Aren't we keep "Hello world" as it is? And it also makes sense and understandable.  "કેમ છો?" is not the exact translation in Gujarati. If we keep "નમસ્તે દુનિયા" then it sounds little weird.

**Comment by ron4ex** [comment]
We've explored many other translations (eiter finished or in progress) and they are going for a native phrase of their language. You can check them out at https://reactjs.org/languages

Harmless strings that are meant just for the purpose of examples can be native and in my opinion should be native. Think about it this way - _Translation of Hello World is an opportunity to add a Gujarati flavor to the documentation. It is relatable to the readers and keeps them excited._ 😃

**Comment by 3sanket3** [comment]
Added Rendering -> રેન્ડરિંગ

**Comment by 3sanket3** [comment]
@Str1kemeister @ron4ex @sonipreshita 

We will have to keep 'Hello World' as it is. At many places in documentation, the code from `.js`  file is directly pulled for code snippets in doc. And if the code in `.js` file tries to print 'Hello World', we will have update the `.js` file from `src` folder. Which we think out of translation scope. We may have to update the same in other translated pages as well.

Thanks all for your inputs. I have updated the Glossary accordingly.

**Comment by 3sanket3** [comment]
Any suggestions for the translation of 'immutable'? 
- અપરિવર્તીત ( as @sonipreshita translated in PR #5 )
- અચળ 

Going with અપરિવર્તીત  for now. I would love to hear your suggestions.

**Comment by Str1kemeister** [comment]
tutorial | ટ્યૂટોરિઅલ
i think it would be great if it was "Namuno" and for "immutable" ,"satat" is also good option.

**Comment by cristatus** [comment]
> Any suggestions for the translation of 'immutable'?
> 
> * અપરિવર્તીત ( as @sonipreshita translated in PR #5 )
> * અચળ
> 
> Going with અપરિવર્તીત for now. I would love to hear your suggestions.

અપરિવર્તીત is more like `unchanged`. I think better word is "અચળ" or "અવિચલ". You can refer http://www.bhagvadgomandal.com/ when any doubts.

**Comment by ron4ex** [comment]
અચળ or સતત are literal translations of the word _constant_ which would be conflicting when we'll need to translate the word _constant_. I think અપરિવર્તીત can be uniquely associated with the word _immutable_.

**Comment by amincharoliya** [comment]
Any suggestions for the word "assert".
The sentence is: "assert facts about what its render method returns".

**Comment by ron4ex** [comment]
I think the phrase that you've mentioned uses the word `assert` in reference to testing. And in testing, assert points to confirming or verifying but in a more intense manner I guess. So something along the lines of `ખાતરીપૂર્વક ચકાસો` might be the closest translation of it.

**Comment by 3sanket3** [comment]
>React also depends on `requestAnimationFrame` (even in test environments).
You can use the [raf](https://www.npmjs.com/package/raf) package to shim `requestAnimationFrame`:

Any input on, what does _shim_ means in above paragraph?

cc: @ron4ex

**Comment by ron4ex** [comment]
`shim` is word used for a wrapper of any sorts. For example, an older browser that doesn't support `fetch` natively may require a shim. `polyfill` is a type os `shim`.

I think we should consider both of those words as JavaScript jargons and use them as they are.

**Comment by 3sanket3** [comment]
Ok, make sense. We will use them as it is(in English itself). Updated the [issue body](https://github.com/reactjs/gu.reactjs.org/issues/6#issue-420723115).

**Comment by kamal250** [comment]
Mock should be `મૉક`.

**Comment by amincharoliya** [comment]
> I think the phrase that you've mentioned uses the word `assert` in reference to testing. And in testing, assert points to confirming or verifying but in a more intense manner I guess. So something along the lines of `ખાતરીપૂર્વક ચકાસો` might be the closest translation of it.

@ron4ex You are right. But after reading the whole paragraph "assert" meaning "ખાતરીપૂર્વક ચકાસો" doesn't fit well. Check the below paragraph:

> When writing unit tests for React, shallow rendering can be helpful. Shallow rendering lets you render a component “one level deep” and assert facts about what its render method returns, without worrying about the behavior of child components, which are not instantiated or rendered. This does not require a DOM."

**Comment by ron4ex** [comment]
I think in this case, it's more about showing the intent rather than testing-related `assert`. It's not necessary to go for literal translation when there isn't a straight forward one. The intent here is clearly about how much **confidence** shallow rendering can give the user. 

How about translating for _...and gives you confidence about what its render method returns..._

**Comment by vsgheravada** [comment]
> > Any suggestions for the translation of 'immutable'?
> > 
> > * અપરિવર્તીત ( as @sonipreshita translated in PR #5 )
> > * અચળ
> > 
> > Going with અપરિવર્તીત for now. I would love to hear your suggestions.
> 
> અપરિવર્તીત is more like `unchanged`. I think better word is "અચળ" or "અવિચલ". You can refer http://www.bhagvadgomandal.com/ when any doubts.

I agree with you. "અચળ" will be more familiar & understandable.

**Comment by harshilparmar** [comment]
@3sanket3 I am working on the hooks introduction page, is "hook" should be translated or not?

**Comment by ron4ex** [comment]
@harshilparmar Yeah I think hooks translation is assigned to you as per #1. The translation progress has been stalled a bit. It would be great if you can put in any page. Please don't be hesitant to create a PR and submit.

**Comment by harshilparmar** [comment]
@ron4ex Thanks brother..I want to ask that is "hook" word going to as it is or it should be translated in gujarati ? like "હૂક"?

**Comment by ron4ex** [comment]
Oh okay. I misunderstood your original comment. It should be transilerated to "હૂક". The general rule is, if the pure gujarati translation doesn't make sense, just transilerate as indicated by some of the examples in issue description.

**Comment by PhoenixCreation** [comment]
What about **react**? Should it be translated to રીએક્ટ ? or should I keep it 'react'?

**Comment by ron4ex** [comment]
Keep _react_ in English.

**Comment by PhoenixCreation** [comment]
ok...Thanks
